### ISSUE TITLE
2021 CWL Mini Conference planning

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/common-workflow-language#919]
- [x] announce shift to February
  - [x] [general mailing list](https://groups.google.com/g/common-workflow-language/c/40TNKYKjnUg/m/miJIt4B1BwAJ)
  - [x] [discourse](https://cwl.discourse.group/t/invite-for-cwl-virtual-miniconference-february-8-10th-2021/253/3)
  - [x] [GitHub Wiki](https://github.com/common-workflow-language/common-workflow-language/wiki/2021-CWL-Mini-Conference)
  - [x] gitter
  - [x] [twitter](https://twitter.com/commonwl/status/1346453606543859715)
  - [x] individual emails to those who RSVPd
- [x] write up instructions for those recording talks
  - [x] what resolution
  - [x] where to upload
    - [x] youtube and something non-Google? (Zenodo?)
  - [x] how to easily add captions
    - [x] upload to youtube, wait for process, edit the auto-generated captions and download them
 - [x] test these instructions, including playback with captions on in Jitsi
 - [x] [Build a schedule from the responses from participants](https://docs.google.com/spreadsheets/d/1mv7hDqVAi9h1LrZjbnehJilJY_YV-HneM2EpxODmJns/edit?usp=sharing)
   - [x] Leave time in the 4 hours for a couple breaks
 - [x] investigate options for socializing sessions
   - [x] https://gather.town/app/HWc7f82Wnd2pudcW/cwl2021
   - [ ] https://cozyroom.xyz/?room=CWL2021
   - [ ] https://topia.io/welcome (doesn't support Firefox)
   - [ ] https://map.highfidelity.com/h8l31Yt5i5lbm1kk/

**Comment by mr-c** [comment]
Here is a summary of the topics that were suggested for discussion (but for which no one has yet to volunteer to lead):

- bio-cwl-tools working session
- Establishing common CWL Best Practice Guides (more of a collaborative discussion session)
  Quickly walk through the multiple guides to revise/join them, including the new BioExcel guides on CWL development and
  engines https://docs.bioexcel.eu/ - then brainstorm if the recommendations are still correct, and what other things the guides
  should cover. Need feedback both from CWL experts and beginners!
- Creating portable CWL code
- Reuse/caching of CWL workflow outputs
- would be interested in learning about best methods for pipeline efficiency especially with large datasets (where we might chunk
  and scatter for example) and understanding error logs.
- CWL workflow engine development
- CWL vs WDL 
- Ideal use cases for CWL, CWL tool box
- GPU support
- CWL integrated with GH Actions.
- workflow development tooling (linting, CI/CD, unit testing or documentation)
- Workflow Engines, (comparison of ?) 
- Data structures for storing CWL

So, how do we fit everything in? :-)

Shall we have break out groups?

Should we tell people to watch the presentations prior and only do Q&A during the sessions?

Presenter's availability across the 3 sessions is skewed towards Session 1: 11 presenters have indicated they are only available for session 1; adding their talk time + 5 minutes for Q&A bring us to 3 hour, 5 minutes out of 4 hours total.

**Comment by mr-c** [comment]
I've started to organize the presenters and their session availabilities into a schedule at https://docs.google.com/spreadsheets/d/1mv7hDqVAi9h1LrZjbnehJilJY_YV-HneM2EpxODmJns/edit?usp=sharing

But we need to decide first about the general format.

**Comment by mr-c** [comment]
I'm going to email speakers who have indicated that they can only speak in Session 1 to confirm and to see if I can move enough of them to Session 2 or 3 so we can have 2 hours of break outs in each session

**Comment by mr-c** [comment]
Okay, I filled up the [draft schedule] for Sessions 2 (6 talks + Q&A) and Session 3 (6 talks + Q&A) leaving two hours at the end for break out sessions.

That leaves 12 presentation proposals for Session 1 (!!!)

Options:
1. Shorten the time for break out discussions to an hour. Convince some of the Session 1 speakers to move to Session 2 or 3.
2. Extend Session 1 by an hour or two, making a total of 5 or 6 hours (with 15-20 minute break in the middle)
3. Schedule "session 1b", same times, but on Thursday, February 11th.
4. ???

Feedback appreciated @inutano @swzCuroverse @tetron

**Comment by mr-c** [comment]
I've also edited the sign up form to not mention giving a talk

**Comment by swzCuroverse** [comment]
Hi all -  For session 1 - I am counting 11, unless I am miscounting.  If we reduce the 15 minute talks to 10 minutes, we get 30 minutes back.  We could do that and have a longer break out session.   And also reason out to the Session 1 speakers about switching slots.  With that we might get even more time back.

**Comment by mr-c** [comment]
@swzCuroverse Yep, when we get the videos I wouldn't be surprised if that frees up some time. Lets make the videos due on January 25th?

**Comment by swzCuroverse** [comment]
OK, sounds good.

**Comment by mr-c** [comment]
> Hi all - For session 1 - I am counting 11, unless I am miscounting. If we reduce the 15 minute talks to 10 minutes, we get 30 minutes back. We could do that and have a longer break out session. And also reason out to the Session 1 speakers about switching slots. With that we might get even more time back.

I was able to put @stain in Session 2 by shortening the time for the intro & CWL 1.2.1 talks and his slot at well (15-> 10 minute for Stian's talk + 5 minutes Q&A).

There's now room for a 5 minute talk + Q&A for Session 3.  @ionox0 , can you give your talk some time between 19:00 - 23:00 PM Eastern USA time on Tuesday, February 10th?

If that works for Ian, then the Session 1 schedule would only run 45 minutes longer than the other days (not changing the length of the break out sessions). That feels okay to me.

**Comment by mr-c** [comment]
Okay, I've added all the talks to the draft schedule. Now we need to email each presenter to confirm the title of their talk, the length/date/time, and to give them upload instructions.

To do that we need to first write + test the upload instructions.

**Comment by mr-c** [comment]
Almost all talks have been uploaded and posted to the forum. I've added the talks to https://github.com/common-workflow-language/common-workflow-language/wiki/2021-CWL-Mini-Conference with links to the forum posts.

I got a professional jisti account as we have over 50 signups for Session 1, and meet.jit.si has a connection limit of 50.

@swzCuroverse @tetron and I are doing a test on Friday and we will send out an email with connection information to all registered attendees (including the speakers) on Friday as well
